[{"authorTime":"2018-11-23 07:19:45","codes":[{"authorDate":"2018-11-23 07:19:45","commitOrder":1,"curCode":"    public void failedActionWithAllowedExceptionWithListenerChangingResultTest() throws Exception {\n        final int ATTEMPTS = 1;\n        final String ACTION = \"print\";\n\n        RingBufferLogHandler handler = new RingBufferLogHandler(20);\n        Logger.getLogger(Retrier.class.getName()).addHandler(handler);\n\n        Retrier<Boolean> r = new Retrier.Builder<>\n                \r\n                (\n                        \r\n                        (Callable<Boolean>) () -> {\n                            throw new IndexOutOfBoundsException(\"Exception allowed considered as failure\");\n                        },\n                        \r\n                        (currentAttempt, result) -> result,\n                        \r\n                        ACTION\n                )\n\n                \r\n                .withAttempts(ATTEMPTS)\n                \r\n                .withDuringActionExceptions(new Class[]{IndexOutOfBoundsException.class})\n                \r\n                .withDuringActionExceptionListener((attempt, exception) -> true)\n                \r\n                .build();\n\n        \r\n        Boolean finalResult = r.start();\n        Assert.assertTrue(finalResult == null ? false : finalResult);\n\n        \r\n        String textSuccess = Messages.Retrier_Success(ACTION, ATTEMPTS);\n        assertTrue(String.format(\"The log should contain '%s'\", textSuccess), handler.getView().stream().anyMatch(m -> m.getMessage().contains(textSuccess)));\n\n        \r\n        String testException = Messages.Retrier_ExceptionFailed(ATTEMPTS, ACTION);\n        assertTrue(String.format(\"The log should contain '%s'\", testException), handler.getView().stream().anyMatch(m -> m.getMessage().startsWith(testException)));\n    }\n","date":"2018-11-23 07:19:45","endLine":233,"groupId":"7125","id":1,"instanceNumber":1,"isCurCommit":0,"methodName":"failedActionWithAllowedExceptionWithListenerChangingResultTest","params":"()","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-jenkins-10-0.7/blobInfo/CC_OUT/blobs/c1/66a54e0b0e8dfb4cb827c8f1d67553976b57be.src","preCode":"    public void failedActionWithAllowedExceptionWithListenerChangingResultTest() throws Exception {\n        final int ATTEMPTS = 1;\n        final String ACTION = \"print\";\n\n        RingBufferLogHandler handler = new RingBufferLogHandler(20);\n        Logger.getLogger(Retrier.class.getName()).addHandler(handler);\n\n        Retrier<Boolean> r = new Retrier.Builder<>\n                \r\n                (\n                        \r\n                        (Callable<Boolean>) () -> {\n                            throw new IndexOutOfBoundsException(\"Exception allowed considered as failure\");\n                        },\n                        \r\n                        (currentAttempt, result) -> result,\n                        \r\n                        ACTION\n                )\n\n                \r\n                .withAttempts(ATTEMPTS)\n                \r\n                .withDuringActionExceptions(new Class[]{IndexOutOfBoundsException.class})\n                \r\n                .withDuringActionExceptionListener((attempt, exception) -> true)\n                \r\n                .build();\n\n        \r\n        Boolean finalResult = r.start();\n        Assert.assertTrue(finalResult == null ? false : finalResult);\n\n        \r\n        String textSuccess = Messages.Retrier_Success(ACTION, ATTEMPTS);\n        assertTrue(String.format(\"The log should contain '%s'\", textSuccess), handler.getView().stream().anyMatch(m -> m.getMessage().contains(textSuccess)));\n\n        \r\n        String testException = Messages.Retrier_ExceptionFailed(ATTEMPTS, ACTION);\n        assertTrue(String.format(\"The log should contain '%s'\", testException), handler.getView().stream().anyMatch(m -> m.getMessage().startsWith(testException)));\n    }\n","realPath":"core/src/test/java/hudson/util/RetrierTest.java","repoName":"jenkins","snippetEndLine":0,"snippetStartLine":0,"startLine":193,"status":"B"},{"authorDate":"2018-11-23 07:19:45","commitOrder":1,"curCode":"    public void failedActionWithAllowedExceptionByInheritanceTest() throws Exception {\n        final int ATTEMPTS = 1;\n        final String ACTION = \"print\";\n\n        RingBufferLogHandler handler = new RingBufferLogHandler(20);\n        Logger.getLogger(Retrier.class.getName()).addHandler(handler);\n\n        Retrier<Boolean> r = new Retrier.Builder<>\n                \r\n                (\n                        \r\n                        (Callable<Boolean>) () -> {\n                            \r\n                            throw new ArrayIndexOutOfBoundsException(\"Unallowed exception breaks the process\");\n                        },\n                        \r\n                        (currentAttempt, result) -> result,\n                        \r\n                        ACTION\n                )\n\n                \r\n                .withAttempts(ATTEMPTS)\n                \r\n                .withDuringActionExceptions(new Class[]{IndexOutOfBoundsException.class})\n                \r\n                .withDuringActionExceptionListener((attempt, exception) -> true)\n                \r\n                .build();\n\n        \r\n        Boolean finalResult = r.start();\n        Assert.assertTrue(finalResult == null ? false : finalResult);\n\n        \r\n        String textSuccess = Messages.Retrier_Success(ACTION, ATTEMPTS);\n        assertTrue(String.format(\"The log should contain '%s'\", textSuccess), handler.getView().stream().anyMatch(m -> m.getMessage().contains(textSuccess)));\n\n        \r\n        String testException = Messages.Retrier_ExceptionFailed(ATTEMPTS, ACTION);\n        assertTrue(String.format(\"The log should contain '%s'\", testException), handler.getView().stream().anyMatch(m -> m.getMessage().startsWith(testException)));\n    }\n","date":"2018-11-23 07:19:45","endLine":277,"groupId":"7125","id":2,"instanceNumber":2,"isCurCommit":0,"methodName":"failedActionWithAllowedExceptionByInheritanceTest","params":"()","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-jenkins-10-0.7/blobInfo/CC_OUT/blobs/c1/66a54e0b0e8dfb4cb827c8f1d67553976b57be.src","preCode":"    public void failedActionWithAllowedExceptionByInheritanceTest() throws Exception {\n        final int ATTEMPTS = 1;\n        final String ACTION = \"print\";\n\n        RingBufferLogHandler handler = new RingBufferLogHandler(20);\n        Logger.getLogger(Retrier.class.getName()).addHandler(handler);\n\n        Retrier<Boolean> r = new Retrier.Builder<>\n                \r\n                (\n                        \r\n                        (Callable<Boolean>) () -> {\n                            \r\n                            throw new ArrayIndexOutOfBoundsException(\"Unallowed exception breaks the process\");\n                        },\n                        \r\n                        (currentAttempt, result) -> result,\n                        \r\n                        ACTION\n                )\n\n                \r\n                .withAttempts(ATTEMPTS)\n                \r\n                .withDuringActionExceptions(new Class[]{IndexOutOfBoundsException.class})\n                \r\n                .withDuringActionExceptionListener((attempt, exception) -> true)\n                \r\n                .build();\n\n        \r\n        Boolean finalResult = r.start();\n        Assert.assertTrue(finalResult == null ? false : finalResult);\n\n        \r\n        String textSuccess = Messages.Retrier_Success(ACTION, ATTEMPTS);\n        assertTrue(String.format(\"The log should contain '%s'\", textSuccess), handler.getView().stream().anyMatch(m -> m.getMessage().contains(textSuccess)));\n\n        \r\n        String testException = Messages.Retrier_ExceptionFailed(ATTEMPTS, ACTION);\n        assertTrue(String.format(\"The log should contain '%s'\", testException), handler.getView().stream().anyMatch(m -> m.getMessage().startsWith(testException)));\n    }\n","realPath":"core/src/test/java/hudson/util/RetrierTest.java","repoName":"jenkins","snippetEndLine":0,"snippetStartLine":0,"startLine":236,"status":"B"}],"commitId":"896356a03893c300361dd8a3acc3b83f926301b7","commitMessage":"@@@Merge commit '8d429ff13b2013ff0b390d59f9c8a7674e6d24a1' into security-master\n","date":"2018-11-23 07:19:45","modifiedFileCount":"5","status":"B","submitter":"Daniel Beck"},{"authorTime":"2021-05-07 16:35:52","codes":[{"authorDate":"2021-05-07 16:35:52","commitOrder":2,"curCode":"    public void failedActionWithAllowedExceptionWithListenerChangingResultTest() throws Exception {\n        final int ATTEMPTS = 1;\n        final String ACTION = \"print\";\n\n        RingBufferLogHandler handler = new RingBufferLogHandler(20);\n        Logger.getLogger(Retrier.class.getName()).addHandler(handler);\n\n        Retrier<Boolean> r = new Retrier.Builder<>\n                \r\n                (\n                        \r\n                        (Callable<Boolean>) () -> {\n                            throw new IndexOutOfBoundsException(\"Exception allowed considered as failure\");\n                        },\n                        \r\n                        (currentAttempt, result) -> result,\n                        \r\n                        ACTION\n                )\n\n                \r\n                .withAttempts(ATTEMPTS)\n                \r\n                .withDuringActionExceptions(new Class[]{IndexOutOfBoundsException.class})\n                \r\n                .withDuringActionExceptionListener((attempt, exception) -> true)\n                \r\n                .build();\n\n        \r\n        Boolean finalResult = r.start();\n        Assert.assertTrue(finalResult != null && finalResult);\n\n        \r\n        String textSuccess = Messages.Retrier_Success(ACTION, ATTEMPTS);\n        assertTrue(String.format(\"The log should contain '%s'\", textSuccess), handler.getView().stream().anyMatch(m -> m.getMessage().contains(textSuccess)));\n\n        \r\n        String testException = Messages.Retrier_ExceptionFailed(ATTEMPTS, ACTION);\n        assertTrue(String.format(\"The log should contain '%s'\", testException), handler.getView().stream().anyMatch(m -> m.getMessage().startsWith(testException)));\n    }\n","date":"2021-05-07 16:35:52","endLine":233,"groupId":"10831","id":3,"instanceNumber":1,"isCurCommit":0,"methodName":"failedActionWithAllowedExceptionWithListenerChangingResultTest","params":"()","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-jenkins-10-0.7/blobInfo/CC_OUT/blobs/a1/3519e1e710b51cc1cd5cae6a4beb07bdc693ee.src","preCode":"    public void failedActionWithAllowedExceptionWithListenerChangingResultTest() throws Exception {\n        final int ATTEMPTS = 1;\n        final String ACTION = \"print\";\n\n        RingBufferLogHandler handler = new RingBufferLogHandler(20);\n        Logger.getLogger(Retrier.class.getName()).addHandler(handler);\n\n        Retrier<Boolean> r = new Retrier.Builder<>\n                \r\n                (\n                        \r\n                        (Callable<Boolean>) () -> {\n                            throw new IndexOutOfBoundsException(\"Exception allowed considered as failure\");\n                        },\n                        \r\n                        (currentAttempt, result) -> result,\n                        \r\n                        ACTION\n                )\n\n                \r\n                .withAttempts(ATTEMPTS)\n                \r\n                .withDuringActionExceptions(new Class[]{IndexOutOfBoundsException.class})\n                \r\n                .withDuringActionExceptionListener((attempt, exception) -> true)\n                \r\n                .build();\n\n        \r\n        Boolean finalResult = r.start();\n        Assert.assertTrue(finalResult == null ? false : finalResult);\n\n        \r\n        String textSuccess = Messages.Retrier_Success(ACTION, ATTEMPTS);\n        assertTrue(String.format(\"The log should contain '%s'\", textSuccess), handler.getView().stream().anyMatch(m -> m.getMessage().contains(textSuccess)));\n\n        \r\n        String testException = Messages.Retrier_ExceptionFailed(ATTEMPTS, ACTION);\n        assertTrue(String.format(\"The log should contain '%s'\", testException), handler.getView().stream().anyMatch(m -> m.getMessage().startsWith(testException)));\n    }\n","realPath":"core/src/test/java/hudson/util/RetrierTest.java","repoName":"jenkins","snippetEndLine":0,"snippetStartLine":0,"startLine":193,"status":"M"},{"authorDate":"2021-05-07 16:35:52","commitOrder":2,"curCode":"    public void failedActionWithAllowedExceptionByInheritanceTest() throws Exception {\n        final int ATTEMPTS = 1;\n        final String ACTION = \"print\";\n\n        RingBufferLogHandler handler = new RingBufferLogHandler(20);\n        Logger.getLogger(Retrier.class.getName()).addHandler(handler);\n\n        Retrier<Boolean> r = new Retrier.Builder<>\n                \r\n                (\n                        \r\n                        (Callable<Boolean>) () -> {\n                            \r\n                            throw new ArrayIndexOutOfBoundsException(\"Unallowed exception breaks the process\");\n                        },\n                        \r\n                        (currentAttempt, result) -> result,\n                        \r\n                        ACTION\n                )\n\n                \r\n                .withAttempts(ATTEMPTS)\n                \r\n                .withDuringActionExceptions(new Class[]{IndexOutOfBoundsException.class})\n                \r\n                .withDuringActionExceptionListener((attempt, exception) -> true)\n                \r\n                .build();\n\n        \r\n        Boolean finalResult = r.start();\n        Assert.assertTrue(finalResult != null && finalResult);\n\n        \r\n        String textSuccess = Messages.Retrier_Success(ACTION, ATTEMPTS);\n        assertTrue(String.format(\"The log should contain '%s'\", textSuccess), handler.getView().stream().anyMatch(m -> m.getMessage().contains(textSuccess)));\n\n        \r\n        String testException = Messages.Retrier_ExceptionFailed(ATTEMPTS, ACTION);\n        assertTrue(String.format(\"The log should contain '%s'\", testException), handler.getView().stream().anyMatch(m -> m.getMessage().startsWith(testException)));\n    }\n","date":"2021-05-07 16:35:52","endLine":277,"groupId":"10831","id":4,"instanceNumber":2,"isCurCommit":0,"methodName":"failedActionWithAllowedExceptionByInheritanceTest","params":"()","path":"/mnt/clonedata/CloneManagementServer/ManagementServer/consistResult/result-jenkins-10-0.7/blobInfo/CC_OUT/blobs/a1/3519e1e710b51cc1cd5cae6a4beb07bdc693ee.src","preCode":"    public void failedActionWithAllowedExceptionByInheritanceTest() throws Exception {\n        final int ATTEMPTS = 1;\n        final String ACTION = \"print\";\n\n        RingBufferLogHandler handler = new RingBufferLogHandler(20);\n        Logger.getLogger(Retrier.class.getName()).addHandler(handler);\n\n        Retrier<Boolean> r = new Retrier.Builder<>\n                \r\n                (\n                        \r\n                        (Callable<Boolean>) () -> {\n                            \r\n                            throw new ArrayIndexOutOfBoundsException(\"Unallowed exception breaks the process\");\n                        },\n                        \r\n                        (currentAttempt, result) -> result,\n                        \r\n                        ACTION\n                )\n\n                \r\n                .withAttempts(ATTEMPTS)\n                \r\n                .withDuringActionExceptions(new Class[]{IndexOutOfBoundsException.class})\n                \r\n                .withDuringActionExceptionListener((attempt, exception) -> true)\n                \r\n                .build();\n\n        \r\n        Boolean finalResult = r.start();\n        Assert.assertTrue(finalResult == null ? false : finalResult);\n\n        \r\n        String textSuccess = Messages.Retrier_Success(ACTION, ATTEMPTS);\n        assertTrue(String.format(\"The log should contain '%s'\", textSuccess), handler.getView().stream().anyMatch(m -> m.getMessage().contains(textSuccess)));\n\n        \r\n        String testException = Messages.Retrier_ExceptionFailed(ATTEMPTS, ACTION);\n        assertTrue(String.format(\"The log should contain '%s'\", testException), handler.getView().stream().anyMatch(m -> m.getMessage().startsWith(testException)));\n    }\n","realPath":"core/src/test/java/hudson/util/RetrierTest.java","repoName":"jenkins","snippetEndLine":0,"snippetStartLine":0,"startLine":236,"status":"M"}],"commitId":"deb3a82d01d8440934a563c492483bbd155fe4ba","commitMessage":"@@@Merge pull request #5458 from StefanSpieker/unneeded_throws\n\n","date":"2021-05-07 16:35:52","modifiedFileCount":"23","status":"M","submitter":"Tim Jacomb"}]
